vizro-core/src/vizro/models/_components/form/date_picker.py

```diff
@@ -0,0 +1,73 @@
+from typing import List, Literal, Optional, Union
+
+import dash_mantine_components as dmc
+from dash import html
+
+try:
+    from pydantic.v1 import Field, PrivateAttr, validator
+except ImportError:  # pragma: no cov
+    from pydantic import Field, PrivateAttr, validator
+
+
+from datetime import date
+
+from vizro.models import Action, VizroBaseModel
+from vizro.models._action._actions_chain import _action_validator_factory
+from vizro.models._components.form._form_utils import validate_max, validate_range_value
+
+
+class DatePicker(VizroBaseModel):
+    """Temporal selector `DatePicker`.
+
+    Can be provided to [`Filter`][vizro.models.Filter] or [`Parameter`][vizro.models.Parameter].
+    Based on the underlying [`dmc.DatePicker`](https://www.dash-mantine-components.com/components/datepicker).
+
+    Args:
+        type (Literal["date_picker"]): Defaults to `"date_picker"`.
+        min (Optional[date]): Start date for date picker. Defaults to `None`.
+        max (Optional[date]): End date for date picker. Defaults to `None`.
+        value (Union[List[date], date]): Default date/dates for date picker. Defaults to `None`.
+        title (str): Title to be displayed. Defaults to `""`.
+        multi (bool): Boolean flag for displaying range picker. Default to `True`.
+        actions (List[Action]): See [`Action`][vizro.models.Action]. Defaults to `[]`.
+
+    """
+
+    type: Literal["date_picker"] = "date_picker"
+    min: Optional[date] = Field(None, description="Start date for date picker.")
+    max: Optional[date] = Field(None, description="End date for date picker.")
+    value: Optional[Union[List[date], date]] = Field(None, description="Default date for date picker")
+    title: str = Field("", description="Title to be displayed.")
+    multi: bool = Field(True, description="Boolean flag for displaying range picker.")
+    actions: List[Action] = []
+
+    _input_property: str = PrivateAttr("value")
+    _set_actions = _action_validator_factory("value")
+
+    # Re-used validators
+    _validate_value = validator("value", allow_reuse=True)(validate_range_value)
+    _validate_max = validator("max", allow_reuse=True)(validate_max)
+
+    def build(self):
+        init_value = self.value or ([self.min, self.max] if self.multi else self.min)  # type: ignore[list-item]
+        if self.multi:
+            date_picker = dmc.DateRangePicker
+        else:
+            date_picker = dmc.DatePicker
+
+        date_picker = date_picker(
```

Comment:
```suggestion
        date_picker_class = dmc.DateRangePicker if self.multi else dmc.DatePicker
        
        date_picker = date_picker_class(
```
Moving the conditional to one line is just a minor style thing. Renaming it important though or you have a variable name clash, which doesn't cause problems right now but is not a good idea as it can lead to super confusing error messages.